dvc/path_info.py

```diff
@@ -253,6 +253,15 @@ def __init__(self, url):
         self.query = p.query
         self.fragment = p.fragment
 
+    def replace(self, path=None):
```

Comment:
Could you  also add a simple unit test to `tests/unit/test_path_info.py`, please?